File: source/python/examples/schedulers/slurm_job_list_getter.py
.. code-tab:: python

    from xenon import Server, PasswordCredential

    with Server() as xenon:
        credential = PasswordCredential(
                username='xenon',
                password='javagat')
        location = 'localhost:10022'
        scheduler = xenon.create_scheduler(
                adaptor='slurm',
                location=location,
                credential=credential)

        for queue_name in scheduler.get_queue_names():
            print("List of jobs in queue {queue_name} for SLURM at {location}"
                  .format(queue_name=queue_name, location=location))
            jobs = scheduler.get_jobs(queue_name)
            if not jobs:
                print("(No jobs)")
            else:
                for job in jobs:
                    print("    {}".format(job))

        scheduler.close()

File: readthedocs/code-tabs/java/src/main/java/nl/esciencecenter/xenon/examples/SlurmJobListGetter.java
package nl.esciencecenter.xenon.examples;

import nl.esciencecenter.xenon.credentials.PasswordCredential;
import nl.esciencecenter.xenon.schedulers.Scheduler;

public class SlurmJobListGetter {

    public static void main(String[] args) throws Exception {

        String adaptor = "slurm";
        String location = "ssh://localhost:10022";

        // make the password credential for user 'xenon'
        String username = "xenon";
        char[] password = "javagat".toCharArray();
        PasswordCredential credential = new PasswordCredential(username, password);

        // create the SLURM scheduler
        Scheduler scheduler = Scheduler.create(adaptor, location, credential);

        // ask SLURM for its queues and list the jobs in each
        for (String queueName : scheduler.getQueueNames()) {
            System.out.println("List of jobs in queue '" + queueName + "' for SLURM at " + location + "");
            String[] jobs = scheduler.getJobs(queueName);
            if (jobs.length == 0) {
                System.out.println("(No jobs)");
            } else {
                for (String job : jobs) {
                    System.out.println(job);
                }
            }
        }
    }
}
